dwave/optimization/src/nodes/mathematical.cpp

```diff
@@ -531,6 +531,272 @@ template class NaryOpNode<functional::min<double>>;
 template class NaryOpNode<std::multiplies<double>>;
 template class NaryOpNode<std::plus<double>>;
 
+// PartialReduceNode *****************************************************************
+template <class BinaryOp>
+struct ExtraArrayData {
+    void commit() {}
+    void revert() {}
+};
```

Comment:
I am guessing this is for future-proofing? Because it's in the `.cpp` file there are no binary compatibility issues so I'd remove it for now. That would also allow you to use `ArrayNodeStateData` directly I believe.

Reply:
yeah, in case of e.g. a product along an axis, then we'd need this extra array data. And it's gotta be an array and not scalars of course. I can remove this extra machinery or keep it commented out for future.

Reply:
I'd remove for now, rather than commenting out. It'll be in the git history if we need it later.